File: Java/108. Convert Sorted Array to Binary Search Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode sol(int l, int r, int[] nums){
        if(l>r){
            return null;
        }
        int m = (l+r)/2;
        TreeNode root= new TreeNode(nums[m]);
        root.left=sol(l,m-1,nums);
        root.right=sol(m+1,r,nums);
        return root;
    }
    public TreeNode sortedArrayToBST(int[] nums) {
        return sol(0,nums.length-1,nums);
    }
}

File: JavaScript/108. Convert Sorted Array to Binary Search Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {number[]} nums
 * @return {TreeNode}
 */
var sol = function(l,r,nums){
    if(l>r){
        return null;
    }
    var m = Math.floor((l+r)/2);
    let root= new TreeNode(nums[m]);
    root.left=sol(l,m-1,nums);
    root.right=sol(m+1,r,nums);
    return root;
}
var sortedArrayToBST = function(nums) {
    return sol(0,nums.length-1,nums);
}
